test/remote/instance.go

```diff
@@ -121,31 +123,47 @@ func (i *InstanceInfo) CreateOrGetInstance(imageURL, serviceAccount string) erro
 		Email:  serviceAccount,
 		Scopes: []string{"https://www.googleapis.com/auth/cloud-platform"},
 	}
-	inst.ServiceAccounts = []*compute.ServiceAccount{saObj}
+	newInst.ServiceAccounts = []*compute.ServiceAccount{saObj}
 
 	if pubkey, ok := os.LookupEnv("JENKINS_GCE_SSH_PUBLIC_KEY_FILE"); ok {
 		klog.V(4).Infof("JENKINS_GCE_SSH_PUBLIC_KEY_FILE set to %v, adding public key to Instance", pubkey)
 		meta, err := generateMetadataWithPublicKey(pubkey)
 		if err != nil {
 			return err
 		}
-		inst.Metadata = meta
+		newInst.Metadata = meta
 	}
 
-	if _, err := i.computeService.Instances.Get(i.project, i.zone, inst.Name).Do(); err != nil {
-		op, err := i.computeService.Instances.Insert(i.project, i.zone, inst).Do()
-		klog.V(4).Infof("Inserted instance %v in project: %v, zone: %v", inst.Name, i.project, i.zone)
-		if err != nil {
-			ret := fmt.Sprintf("could not create instance %s: API error: %v", i.name, err)
-			if op != nil {
-				ret = fmt.Sprintf("%s. op error: %v", ret, op.Error)
+	curInst, _ := i.computeService.Instances.Get(i.project, i.zone, newInst.Name).Do()
+	if curInst != nil {
+		if !strings.Contains(curInst.MachineType, newInst.MachineType) {
```

Comment:
Can you just compare these as equal?

Reply:
The machine type string get from service has a prefix [https://www.googleapis.com/compute/v1/projects/${PROJECT}](https://www.googleapis.com/compute/v1/projects/$%7BPROJECT%7D). If we want to use equal I need to parse the string. I feel parse prefix is similar to using Contains?

Reply:
Ok, let's keep as is then.